controllers/front/CartController.php

```diff
@@ -635,16 +675,67 @@ protected function areProductsAvailable()
         }
 
         if ($product['active']) {
+            // get the configuration about showing quantities
+            $display_quantities = Configuration::get('PS_DISPLAY_QTIES');
+
+            // Check if the display quantities option is enabled, and check quantities to provide singular or plural form as well
+            if ($display_quantities) {
+                if ((int) $product['quantity_available'] > 1) {
+                    return $this->trans(
+                        'There are only [1]%quantity%[/1] items of the product [1]%product%[/1] left in stock. Adjust the quantity in your cart to proceed with your order.',
+                        [
+                            '%product%' => $product['name'] . (isset($product['attributes']) && !empty($product['attributes']) ? ' (' . $product['attributes'] . ')' : ''),
+                            '%quantity%' => $product['quantity_available'],
+                            '[1]' => '<strong>',
+                            '[/1]' => '</strong>',
+                        ],
+                        'Shop.Notifications.Error'
+                    );
+                } elseif ((int) $product['quantity_available'] == 1) {
+                    return $this->trans(
+                        'There\'s only [1]%quantity%[/1] item of the product [1]%product%[/1] left in stock. Adjust the quantity in your cart to proceed with your order.',
+                        [
+                            '%product%' => $product['name'] . (isset($product['attributes']) && !empty($product['attributes']) ? ' (' . $product['attributes'] . ')' : ''),
+                            '%quantity%' => $product['quantity_available'],
+                            '[1]' => '<strong>',
+                            '[/1]' => '</strong>',
+                        ],
+                        'Shop.Notifications.Error'
+                    );
+                }
```

Comment:
```suggestion
                return $this->trans(
                    'The quantity of the product [1]%product%[/1] currently available for purchase is [1]%quantity%[/1]. Please adjust the quantity in your cart to proceed with your order.',
                    [
                        '%product%' => $product['name'] . ($product['attributes'] ? ' (' . $product['attributes'] . ')' : ''),
                        '%quantity%' => $product['quantity_available'],
                        '[1]' => '<strong>',
                        '[/1]' => '</strong>',
                    ],
                    'Shop.Notifications.Error'
                );
```
No need for multiple forms :)